examples/ct/ct_server/main.go

```diff
@@ -28,32 +28,57 @@ import (
 	etcdnaming "github.com/coreos/etcd/clientv3/naming"
 	"github.com/golang/glog"
 	"github.com/google/trillian"
+	"github.com/google/trillian/configs"
 	"github.com/google/trillian/examples/ct"
 	"github.com/google/trillian/util"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/naming"
 )
 
-// Global flags that affect all log instances.
-var (
-	serverHostFlag    = flag.String("host", "localhost", "Address to serve CT log requests on")
-	serverPortFlag    = flag.Int("port", 6962, "Port to serve CT log requests on")
-	rpcBackendFlag    = flag.String("log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
-	rpcDeadlineFlag   = flag.Duration("rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
-	logConfigFlag     = flag.String("log_config", "", "File holding log config in JSON")
-	maxGetEntriesFlag = flag.Int64("maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
-	etcdServers       = flag.String("etcd_servers", "", "A comma-separated list of etcd servers")
-)
+type config struct {
+	Host                 string
+	Port                 int
+	LogRPCServer         string
+	RPCDeadline          configs.JSONDuration
+	LogConfig            string
+	MaxGetEntriesAllowed int64
+	EtcdServers          string
+}
+
+var configFlag = flag.String("config", "", "Path to JSON config file to load instead of providing config via flags, if provided may only be used along with the following flags: --logtostderr, --alsologtostderr, --stderrthreshold, --log_dir, --log_backtrace_at, --v, --vmodule")
 
 func main() {
+	c := config{}
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flags.StringVar(&c.Host, "host", "localhost", "Address to serve CT log requests on")
+	flags.IntVar(&c.Port, "port", 6962, "Port to serve CT log requests on")
+	flags.StringVar(&c.LogRPCServer, "log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
+	flags.DurationVar(&c.RPCDeadline.Duration, "rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
+	flags.StringVar(&c.LogConfig, "log_config", "", "File holding log config in JSON")
+	flags.Int64Var(&c.MaxGetEntriesAllowed, "maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
+	flags.StringVar(&c.EtcdServers, "etcd_servers", "", "A comma-separated list of etcd servers")
+	// Both the default flag set and the binary specific flag set need to have Parse() called
+	// so that we can also initialize the flags registered by github.com/golang/glog as well
+	// as the ones defined above
+	flags.Parse(os.Args[1:])
 	flag.Parse()
 
-	if *maxGetEntriesFlag > 0 {
-		ct.MaxGetEntriesAllowed = *maxGetEntriesFlag
+	if *configFlag != "" {
+		// Only allow either --config OR any other flags
+		if flags.NFlag() > 1 {
```

Comment:
Could we move the --config "exclusiveness" check to the shared function as well? (Whatever form it might take, it is.)